### PR TITLE
Fix broken links

### PR DESCRIPTION
**Pull request by shiltemann**
This re-enables link checking and fixes a bunch of broken links.

There are currently 4 broken links remaining, all pointing to hub, and I'm not sure what to do with those:

https://galaxyproject.org/schema/SchemaSpy/index.html
https://galaxyproject.org/duplex/fromReads.png
https://galaxyproject.org/duplex/fromDCS.png
https://galaxyproject.org/duplex/histItem.png

anybody know the correct links for these?

ping @nekrut those last three are images for one of your tutorials (dunovo), do you happen to have those images locally that we could just add them directly to this repo?

also, let me know if I missed any broken links :)

### CODE REVIEW

**Comment by bgruening** [comment]
ping @dannon - the links should be redirected via nginx I suppose?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -346,7 +346,7 @@ logo: "GTN"
 <!-- add a resized image in percentage: 10%, 25%, 50%, or 75% -->
 .image-25[![](../images/ecker_2012.jpg)]
 
-<small>[*Ecker et al, Nature, 2012*](http://www.nature.com/nature/journal/v489/n7414/full/489052a.html)</small>
+<small>[*Ecker et al, Nature, 2012*](https://www.ncbi.nlm.nih.gov/pubmed/22955614)</small>
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: https://doi.org/10.1038/489052a maybe?

File: topics/introduction/tutorials/options-for-using-galaxy/slides.html [hunk 2]
```diff
@@ -139,7 +139,7 @@
 
 ### Commercial support
 
-- You can buy a [preconfigured Galaxy server](https://bioteam.net/products/galaxy-appliance/) from [BioTeam](https://bioteam.net)
+- You can buy a [preconfigured Galaxy server](http://bioteam.net/products/galaxy-appliance/) from [BioTeam](http://bioteam.net/)
```
Comment (nsoranzo) [inline topics/introduction/tutorials/options-for-using-galaxy/slides.html]: These 2 redirect to https for me.

Comment (shiltemann) [inline topics/introduction/tutorials/options-for-using-galaxy/slides.html]: ah yes, meant to revert that, nice catch, thanks for checking all this :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (shiltemann) [inline CONTRIBUTING.md]: for me that redirects to the nature URL and is still giving me 401 in the travis check

Comment (nsoranzo) [inline CONTRIBUTING.md]: Maybe the Nature paywall returns an HTTP error?!? Whatever, not important.

Comment (shiltemann) [inline CONTRIBUTING.md]: yes exactly :)

File: topics/dev/tutorials/tool-integration/slides.html [hunk 3]
```diff
@@ -585,7 +585,7 @@
 $ planemo serve
 ```
 
-Open http://127.0.0.1:9090/ in your web browser to view your new tool
+Open https://127.0.0.1:9090/ in your web browser to view your new tool
```
Comment (nsoranzo) [inline topics/dev/tutorials/tool-integration/slides.html]: I don't think Planemo serves https.

File: topics/dev/tutorials/tool-integration/slides.html [hunk 4]
```diff
@@ -1389,7 +1389,7 @@
 
 ```yaml
 api_key: admin
-galaxy_instance: http://127.0.0.1:8080/
+galaxy_instance: https://127.0.0.1:8080/
```
Comment (nsoranzo) [inline topics/dev/tutorials/tool-integration/slides.html]: This one too.

File: topics/dev/tutorials/architecture/slides.html [hunk 5]
```diff
@@ -1320,7 +1320,7 @@
 .code[```
 galaxy.queue_worker INFO 2016-06-23 19:13:39,049 Binding and starting galaxy control worker for main
 Starting server in PID 21102.
-serving on http://127.0.0.1:8080
+serving on https://127.0.0.1:8080
```
Comment (nsoranzo) [inline topics/dev/tutorials/architecture/slides.html]: And this too.

File: topics/dev/tutorials/interactive-environments/slides.html [hunk 6]
```diff
@@ -423,8 +423,8 @@
     # Note the trailing slash used everywhere!
     location PROXY_PREFIX/helloworld/ {
         proxy_buffering off;
-        proxy_pass         http://127.0.0.1:8000/;
-        proxy_redirect     http://127.0.0.1:8000/ PROXY_PREFIX/helloworld/;
+        proxy_pass         https://127.0.0.1:8000/;
+        proxy_redirect     https://127.0.0.1:8000/ PROXY_PREFIX/helloworld/;
```
Comment (nsoranzo) [inline topics/dev/tutorials/interactive-environments/slides.html]: Here too.

File: topics/usegalaxy/tutorials/rb-rnaseq/tutorial.md [hunk 7]
```diff
@@ -144,25 +144,25 @@ To further optimize and speed up spliced read alignment Kim at al. [2015](https:
 
 ### STAR mapper
 
-[STAR aligner](https://github.com/alexdobin/STAR) is a fast alternative for mapping RNAseq reads against genome utilizing uncompressed [suffix array](https://en.wikipedia.org/wiki/Suffix_array). It operates in [two stages](http://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635.abstract). In the first stage it performs seed search:
+[STAR aligner](https://github.com/alexdobin/STAR) is a fast alternative for mapping RNAseq reads against genome utilizing uncompressed [suffix array](https://en.wikipedia.org/wiki/Suffix_array). It operates in [two stages](https://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635.abstract). In the first stage it performs seed search:
```
Comment (nsoranzo) [inline topics/usegalaxy/tutorials/rb-rnaseq/tutorial.md]: https://academic.oup.com/bioinformatics/article/29/1/15/272537/STAR-ultrafast-universal-RNA-seq-aligner

File: topics/usegalaxy/tutorials/non-dip/tutorial.md [hunk 8]
```diff
@@ -137,7 +137,7 @@ In other words the two datasets had ~6% and ~9% duplicates, respectively.
 
 # Left-aligning indels
 
-Left aligning of indels (a variant of re-aligning) is extremely important for obtaining accurate variant calls. This concept, while not difficult, requires some explanation. For illustrating how left-aligning works we expanded on an example provided by [Tan:2015](http://bioinformatics.oxfordjournals.org/content/31/13/2202.abstract). Suppose you have a reference sequence and a sequencing read:
+Left aligning of indels (a variant of re-aligning) is extremely important for obtaining accurate variant calls. This concept, while not difficult, requires some explanation. For illustrating how left-aligning works we expanded on an example provided by [Tan:2015](https://bioinformatics.oxfordjournals.org/content/31/13/2202.abstract). Suppose you have a reference sequence and a sequencing read:
```
Comment (nsoranzo) [inline topics/usegalaxy/tutorials/non-dip/tutorial.md]: https://academic.oup.com/bioinformatics/article/31/13/2202/196142/Unified-representation-of-genetic-variants

File: topics/usegalaxy/tutorials/ngs/tutorial.md [hunk 9]
```diff
@@ -204,11 +204,11 @@ QC datasets you have uploaded before.
...
 
-- 2009 Bowtie 1 - [Langmead et al.](http://genomebiology.com/content/10/3/R25)
+- 2009 Bowtie 1 - [Langmead et al.](https://genomebiology.com/content/10/3/R25)
 - 2012 Bowtie 2 - [Langmead and Salzberg](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3322381/)
-- 2009 BWA - [Li and Durbin](http://bioinformatics.oxfordjournals.org/content/25/14/1754.long)
-- 2010 BWA - [Li and Durbin](http://bioinformatics.oxfordjournals.org/content/26/5/589)
-- 2013 BWA-MEM - [Li](http://arxiv.org/abs/1303.3997)
+- 2009 BWA - [Li and Durbin](https://bioinformatics.oxfordjournals.org/content/25/14/1754.long)
+- 2010 BWA - [Li and Durbin](https://bioinformatics.oxfordjournals.org/content/26/5/589)
+- 2013 BWA-MEM - [Li](https://arxiv.org/abs/1303.3997)
```
Comment (nsoranzo) [inline topics/usegalaxy/tutorials/ngs/tutorial.md]: 2009 BWA is now https://academic.oup.com/bioinformatics/article/25/14/1754/225615/Fast-and-accurate-short-read-alignment-with
2010 BWA is now https://academic.oup.com/bioinformatics/article/26/5/589/211735/Fast-and-accurate-long-read-alignment-with-Burrows

File: topics/usegalaxy/tutorials/ngs/tutorial.md [hunk 10]
```diff
@@ -42,18 +42,18 @@ Finally, datasets can be uploaded directly from NCBI's short read archive:
...
+[FastQ](https://en.wikipedia.org/wiki/FASTQ_format) is not a very well defined format. In the beginning various manufacturers of sequencing instruments were free to interpret fastq as they saw fit, resulting in a multitude of fastq flavors. This variation stemmed primarily from different ways of encoding quality values as described [here](http://en.wikipedia.org/wiki/FASTQ_format) (below you will explanation of quality scores and their meaning). Today, [fastq Sanger](http://www.ncbi.nlm.nih.gov/pubmed/20015970) version of the format is considered to be the standard form of fastq. Galaxy is using fastq sanger as the only legitimate input for downstream processing tools and provides [a number of utilities for converting fastq files](http://www.ncbi.nlm.nih.gov/pubmed/20562416) into this form (see **NGS: QC and manipulation** section of Galaxy tools).
```
Comment (nsoranzo) [inline topics/usegalaxy/tutorials/ngs/tutorial.md]: http://www.ncbi.nlm.nih.gov/pubmed/20562416 can also be updated to https

File: topics/dev/tutorials/tool-integration/slides.html (hunk 3 above)
Comment (shiltemann) [inline topics/dev/tutorials/tool-integration/slides.html]: thanks! changed all the localhost links back, but forgot about the 127.0.0.1, nice catch again :)

File: topics/usegalaxy/tutorials/ngs/tutorial.md (hunk 9 above)
Comment (shiltemann) [inline topics/usegalaxy/tutorials/ngs/tutorial.md]: awesome, thanks :)

**Comment by bgruening** [comment]
@shiltemann @nsoranzo I'm fine with merging this one before there are merge conflicts. It's already great - thanks a lot Saskia for doing this.
For the 4 remaining links we should create an issue and ping @nekrut and @dannon again.

**Comment by martenson** [comment]
The images are missing from the Main dunovo wiki too (https://github.com/galaxyproject/dunovo/wiki) ping @NickSto

**Comment by nsoranzo** [comment]
> @shiltemann @nsoranzo I'm fine with merging this one before there are merge conflicts. It's already great - thanks a lot Saskia for doing this.

:+1: (anyway I won't have time to contribute to this till the end of the month)

**Comment by bebatut** [comment]
Thanks a lot @shiltemann !!!!!!

**Comment by hrhotz** [comment]
wrt the broken link for the database schema (https://galaxyproject.org/schema/SchemaSpy/index.html): I have contacted @tnabtaf already during GCC. He promissed to look into it.

**Comment by hrhotz** [comment]
also, one of the http to https coversions results in a broken link:  https://schemaspy.sourceforge.net/
(will make a new pull request while fixing a typo as well)

**Comment by shiltemann** [comment]
thanks @hrhotz! 

huh, looks like that link wasn't picked up by the checker, I wonder why..

**Comment by shiltemann** [comment]
ok, maybe need to run the checker in verbose mode, that at least gave an error message for the link, even if still exited successfully

```
HTTPSConnectionPool(host='schemaspy.sourceforge.org', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f9fc6a444a8>: Failed to establish a new connection: [Errno -2] Name or service not known',))
```

**Comment by shiltemann** [comment]
hmm, there are a bunch more silently failing links, fixing now

**Pull request merged**